src/screens/General/Dashboard.js

```diff
@@ -273,226 +264,242 @@ class Dashboard extends Component {
 				{ this.renderFooter() }
 			</View>
 		);
 	}
 
 	renderFooter() {
+		const {
+			footer,
+			footerText,
+			black,
+			textColor
+		} = styles;
+
 		return (
-			<View style = {{ flex: 0.3, justifyContent: "center", backgroundColor: "#FECB00", width: "100%" }}>
-				<View style = {{ flexDirection: "row", justifyContent: "center" }}>
-					<Text style = {{ color: "black" }}>SHPE </Text>
-					<Text style = {{ color: "white" }}>UCF</Text>
+			<View style = { footer }>
+				<View style = { footerText }>
+					<Text style = { black }>SHPE </Text>
+					<Text style = { textColor }>UCF</Text>
 				</View>
 			</View>
 		);
 	}
 
 	renderFlags() {
-		const flagHeight = dimension.height - 0.3 * dimension.height;
+		const {
+			flagModal,
+			flagColumn
+		} = styles;
+
+		const flagHeight = { height: dimension.height - 0.3 * dimension.height };
 		const countriesL = ["AR", "BO", "BR", "CL", "CO", "CR", "CU", "DO", "EC", "SV", "GQ", "GT", "HN"];
 		const countriesR = ["MX", "NI", "PA", "PY", "PE", "PR", "RO", "ES", "TT", "US", "UY", "VE", ""];
 
 		return (
-			<Modal visible = { this.state.flags } transparent = { true }>
-				<SafeAreaView
-					style = {{ position: "absolute", flexDirection: "row", width: dimension.width, height: flagHeight, justifyContent: "space-between", top: dimension.height * 0.15, paddingLeft: "2%", paddingRight: "2%" }}>
-					<View style = {{ justifyContent: "space-evenly", borderColor: "white" }}>
-						{ countriesL.map(item =>
-							<TouchableOpacity onPress = { () => this.flagPicked(item) }>
-								<Flag
-									type = "flat"
-									code = { item }
-									size = { 32 }
-								/>
-							</TouchableOpacity>
-						) }
-					</View>
-					<View style = {{ justifyContent: "space-evenly", borderColor: "white" }}>
-						{ countriesR.map(item =>
-							<TouchableOpacity onPress = { () => this.flagPicked(item) }>
-								<Flag
-									type = "flat"
-									code = { item }
-									size = { 32 }
-								/>
-							</TouchableOpacity>
-						) }
-					</View>
+			<Modal visible = { this.state.flagsVisible } transparent = { true }>
+				<SafeAreaView >
+					<TouchableOpacity
+						style = { [flagModal, flagHeight] }
+						onPress = { () => this.setState({ flagsVisible: false }) }
+					>
+						<View style = { flagColumn }>
+							{ countriesL.map(item =>
+								<TouchableOpacity onPress = { () => this.flagPicked(item) }>
+									<Flag
+										type = "flat"
+										code = { item }
+										size = { 32 }
+									/>
+								</TouchableOpacity>
+							) }
+						</View>
+						<View style = { flagColumn }>
+							{ countriesR.map(item =>
+								<TouchableOpacity onPress = { () => this.flagPicked(item) }>
+									<Flag
+										type = "flat"
+										code = { item }
+										size = { 32 }
+									/>
+								</TouchableOpacity>
+							) }
+						</View>
+					</TouchableOpacity>
 				</SafeAreaView>
 			</Modal>
 		);
 	}
 
-	renderNewFlag() {
+	rendercustomFlag() {
```

Comment:
see above

Reply:
Some of the functionality has been transferred out